restorers/model/nafnet/nafblock.py

```diff
@@ -0,0 +1,151 @@
+from typing import Optional
+
+import tensorflow as tf
+from tensorflow import keras
+
+
+class SimpleGate(keras.layers.Layer):
+    """
+    Simple Gate
+    It splits the input of size (b,h,w,c) into tensors of size (b,h,w,c//factor) and returns their Hadamard product
+    Parameters:
+        factor: the amount by which the channels are scaled down
+    """
+
+    def __init__(self, factor: Optional[int] = 2, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.factor = factor
+
+    def call(self, x: tf.Tensor, *args, **kwargs) -> tf.Tensor:
+        x = tf.expand_dims(x, axis=-1)
+        return tf.reduce_prod(
+            tf.concat(tf.split(x, num_or_size_splits=self.factor, axis=-2), axis=-1),
+            axis=-1,
+        )
+
```

Comment:
Could you add a `get_config` fn to all custom layers ? It helps with distributed training in some instances. [Reference Implementation](https://github.com/SauravMaheshkar/Knowledge-Distillation/blob/ef5b92bd53852df5d920746ba19d870360467e5b/src/backend/tensorflow/modules/affine.py#L46)

Reply:
Could you please explain a bit, what get_config does? Like what class members am I supposed to add to get_config? Is it just the constructor arguments ?